Wikipedia/Code/ArticleFetcher.swift

```diff
@@ -281,6 +281,54 @@ final public class ArticleFetcher: Fetcher, CacheFetching {
         }
     }
     
+    /// Makes periodic HEAD requests to the mobile-html endpoint until the etag no longer matches the one provided.
+    @discardableResult public func waitForMobileHTMLChange(articleURL: URL, eTag: String, timeout: TimeInterval, cancellationKey: CancellationKey? = nil, completion: @escaping (Result<String, Error>) -> Void) -> CancellationKey? {
+        guard timeout > 0 else {
+            completion(.failure(RequestError.timeout))
+            return nil
+        }
+        let requestURL: URL
+        do {
+            requestURL = try mobileHTMLURL(articleURL: articleURL)
+        } catch let error {
+            completion(.failure(error))
+            return nil
+        }
+        let start = CFAbsoluteTimeGetCurrent()
+        let key = cancellationKey ?? UUID().uuidString
+        let maybeTask = session.dataTask(with: requestURL, method: .head, headers: [URLRequest.ifNoneMatchHeaderKey: eTag]) { (_, response, error) in
+            defer {
+                self.untrack(taskFor: key)
+            }
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+            guard
+                let httpURLResponse = response as? HTTPURLResponse,
+                httpURLResponse.statusCode == 200,
```

Comment:
Why check for 200 here? If we get an error it will be caught above. 

It seems like the first three items in this guard are good safety checks - but if they resolve to `false` we probably want to treat them differently than the final `updatedETag != eTag`. The first three should never be false (I think), so if any of them are we might want to failure out of this loop. Let's say there is some weird error, and the server is returning a response w/ no error but including a 500 - the user will be caught in an infinite loop (until the timeout, that is).

Reply:
Good question - the `If-None-Match` header is set above meaning that we should see a response code of 304 (not modified) until the eTag changes. However, there was a bug in this PR causing a cached response to be returned with a 200 status code and the old ETag. I pushed a change to fix that problem by utilizing the `.reloadIgnoringLocalCacheData` cache policy. I also updated the status code handling to fail on any code other than 200 and 304 that don't cause an error to be returned by the networking stack.

Reply:
Looks great!